Src/IronPython.Modules/resource.cs

```diff
@@ -0,0 +1,427 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Numerics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+
+using Microsoft.Scripting.Runtime;
+
+using IronPython.Runtime;
+using IronPython.Runtime.Exceptions;
+using IronPython.Runtime.Operations;
+using IronPython.Runtime.Types;
+
+[assembly: PythonModule("resource", typeof(IronPython.Modules.PythonResourceModule), PlatformsAttribute.PlatformFamily.Unix)]
+namespace IronPython.Modules;
+
+
+[SupportedOSPlatform("linux")]
+[SupportedOSPlatform("macos")]
+public static class PythonResourceModule {
+    public const string __doc__ = "Provides basic mechanisms for measuring and controlling system resources utilized by a program. Unix only.";
+
+    [Obsolete("Deprecated in favor of OSError.")]
```

Comment:
Does having the Obsolete tag produce any side effects in Python?

Reply:
I haven't noticed any and I expect not. In all cases I saw `resource` used, the code was catching `OSError` rather than `resource.error`. It is here just for compatibility, but Python docs call it deprecated. There is a subtle difference between CPython and IronPython here, e.g.: in CPython, `resource.error(2)` creates `OSError`, but in IronPython it creates `FileNotFoundError`. The same applies to `OSError(2)` by the way. Not a big deal, I think.

Reply:
I meant side effects as in something like a `DeprecationWarning`. Not finding a `Obsolete` attribute on a property to, but `clr.GetDynamicType` appears to raise the warning. Not a big deal I guess...

> in IronPython it creates `FileNotFoundError`

Ah interesting. Seems like a bug - only the 2 argument OSError constructor returns a subclass (e.g. `OSError(2, '')` becomes `FileNotFoundError` whereas `OSError(2)` should be `OSError`).